CMakeLists.txt

```diff
@@ -8,51 +8,96 @@ set(CMAKE_TOOLCHAIN_FILE ${CMAKE_SOURCE_DIR}/cmake/stm32-toolchain.cmake)
 project(ATeam-Firmware
 	LANGUAGES C CXX ASM)
 
+macro(tests dir linker fname tcomment prefix)
+	set(TEST_DIR ${dir})
+
+	file(GLOB_RECURSE TEST_ASM_FILES "${TEST_DIR}/*.s")
+	file(GLOB_RECURSE TEST_C_FILES "${TEST_DIR}/*.c")
+	file(GLOB_RECURSE TEST_CXX_FILES "${TEST_DIR}/*.cpp")
+
+	add_executable(${fname}.elf
+		${TEST_ASM_FILES}
+		${TEST_C_FILES}
+		${TEST_CXX_FILES}
+	)
+
+	header_directories(TEST_INCLUDE_DIRS ${TEST_DIR})
+	target_include_directories(${fname}.elf PRIVATE
+		${TEST_INCLUDE_DIRS}
+	)
+
+	target_compile_definitions(${fname}.elf PRIVATE
+		${${prefix}_DEFINITIONS}
+	)
+
+	target_compile_options(${fname}.elf PRIVATE
+		$<$<COMPILE_LANGUAGE:C>:${${prefix}_C_OPTIONS}>
+		$<$<COMPILE_LANGUAGE:CXX>:${${prefix}_CXX_OPTIONS}>
+	) 
+
+	set(test_linker_file "${CMAKE_CURRENT_SOURCE_DIR}/${TEST_DIR}/${linker}")
+	target_link_options(${fname}.elf PRIVATE ${${prefix}_LINKER_OPTIONS} -T ${test_linker_file})
+		
+	#set_target_properties(test-breadboard-radio.elf PROPERTIES
+	#	LINK_DEPENDS ${bb_radio_linker_file})
+
+	add_custom_target(${fname}
+		arm-none-eabi-objcopy -Obinary "${fname}.elf" "${fname}.bin"
+		WORKING_DIRECTORY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}"
+		DEPENDS ${fname}.elf
+		COMMENT ${tcomment}
+	)
+
+	add_custom_target(${fname}-prog
+		openocd -f ${${prefix}_OPENOCD_CFG} -c "program bin/${fname}.bin verify reset exit ${STM32_FLASH_START_ADDR}"
+	)
+endmacro()
+
 #######################################
 #  Standalone Breadbaord Radio Tests  #
 #######################################
```

Comment:
this seems to be passing testing, so we can just nuke the commented out old stuff